src/common/src/config.rs

```diff
@@ -253,6 +253,10 @@ pub struct MetaConfig {
     #[serde(default)]
     pub do_not_config_object_storage_lifecycle: bool,
 
+    /// Whether the endpoint of object storage is force path style.
+    #[serde(default)]
+    pub is_force_path_style: bool,
```

Comment:
There are two issues here:
1. This config is only used by meta and not used by CN/Compactor so CN/Compactor may still fail to connect to object store when the endpoint is path style.
2. This config can only be set via toml so it is unfriendly to cloud deployment when user wants to modify it.

To address the above issue, I think there are two options:
1. Make `is_force_path_style` a mutable system parameter and part of the `OverrideConfigOpts` in MetaNodeopts.
2. Make `is_force_path_style` a env for Compactor/CN/Meta just like `RW_S3_ENDPOINT`. We can add some docs to remind user to set it when there is a failure in connecting to the object store.

Although originally we thought only setting it in meta is better, now after a second thought, I think 2 is better since this config is tightly coupled with `RW_S3_ENDPOINT`. It is somehow strange to have different mechanism to set `RW_S3_ENDPOINT` and `is_force_path_style`.

Reply:
Agree, let's make it the same with RW_S3_ENDPOINT as an env, thus we don't need to modify so many files. It is really not worth it for such a small feature.